### ISSUE TITLE
[Bug] Crash on Startup with Framework

### ISSUE THREAD

**Issue by suerion** [TheMisterFish/Fabric_OfflinePlayersReworked#33]
**Describe the bug**
After Adding the Mod to my huge Modpack, it gives me an error with the library "Framework"

**To Reproduce**
_Steps to reproduce the behavior:_
1. Add the Mod to Modpack
2. Start the Server
3. Server crash on Startup

**Expected behavior**
Should load with offline Players

**Screenshots**

**Versions (please complete the following information):**
- Minecraft: 1.21.1
- Fabric Loader: 0.16.5
- Mod Version: newest

**Additional context**
https://mclo.gs/egIpY3t

**Comment by TheMisterFish** [comment]
After having a quick glance over the log file I don't directly see anything that indicated that this issue is because of OfflinePlayers. Would it be possible for you to send the mods folder to me so I can test it on my machine?

I'll have a better look at the log file as soon as possible.

**Comment by suerion** [comment]
Hey, i'm at work, i will send the whole modpack to you, if i had time.

Will it send to you as a whole zip file ;)

**Comment by suerion** [comment]
Please don't use the Modpack to upload it somewhere, there are mods from exclusive curseforge and some mods exclusive modrinth, its a little bit weird....

and it is only an dev enviroment to fix a lots of errors.

if you would join you need only automodpack in the fabric client and then only try to connect to the server, the server would send the client the mods 

offlineplayersreworked is only as disabled file, you only need to rename it

Zip file has complete modpack server pack.
https://we.tl/t-QIAVrnwN0O

these are the latest log without offline player
https://mclo.gs/ysyEA4T 
[latest.log](https://github.com/user-attachments/files/17284559/latest.log)


Receip errors and command errors you could ignore, it would be fixed in my next update

**Comment by TheMisterFish** [comment]
I might have found a possible problem. Am looking into a fix right now.

(Ps. I deleted your download link after downloading as you said it shouldn't be uploaded :))

**Comment by TheMisterFish** [comment]
I'm not able to run the server so I'm not sure if I fixed the issue or not. Before I make any new releases, could I ask you to test it out @suerion ?

Here's the new  dev-jar: https://www.mediafire.com/file/oe62c7v9wxhdmye/offlineplayersreworked-1.0.9-dev.jar/file

**Comment by TheMisterFish** [comment]
I'm unable to start up the server correctly, but I was able to see a different error when offlinePlayers was installed vs when it wasn't installed. I've updated the .jar building so that it shouldn't interfere with any other mods dependencies but I'm unable to test if it is working correctly. @suerion Could you test it out before I release a new version?

https://www.mediafire.com/file/oe62c7v9wxhdmye/offlineplayersreworked-1.0.9-dev.jar/file

**Comment by suerion** [comment]
Good Morning, sorry, yesterday was a long day. 

I will trying to test it after work. 

And thanks for deleting the link.

Could be, that the you need to change the bat file.

I had an direct link to my jdk21. With another jdk, lower than 21 it would not work, because 1.21.1 need the 21 version.

Thanks for your work.

**Comment by suerion** [comment]
@TheMisterFish 
i think you had send me an forge dev mod file or something, could not test sorry
```
[16:45:49] [main/WARN]: Found 1 non-fabric mod:
        - offlineplayersreworked-1.0.9-dev.jar
```

**Comment by TheMisterFish** [comment]
Wrong file indeed, I'll have to send you the correct one but am currently busy with work. Wil try and send correct file asap.

**Comment by suerion** [comment]
> Wrong file indeed, I'll have to send you the correct one but am currently busy with work. Wil try and send correct file asap.

yeah, no stress, had more time as yesterday :)

**Comment by TheMisterFish** [comment]
> > Wrong file indeed, I'll have to send you the correct one but am currently busy with work. Wil try and send correct file asap.
> 
> yeah, no stress, had more time as yesterday :)

I found the issue. I'm using a dependency called jsonDB as a small database for the offline players. The jsonDB library has a dependency on Reflections. `com.mrcrayfish.framework.FrameworkFabric` also has a dependency on Reflections. This is causing the error.

I was trying to create the jar that includes the jsonDB library with a own Reflections in the jar so they shouldn't clash, but I'm having issues building it correctly (as you saw)

I now feel like I have 3 options:
1. continue trying with shadowJar
2. remove jsonDB and create a custom json database class with no dependencies
3. call it a day... (won't do this one though 😅 )

I will keep you updated in this thread if that's not an issue for you, but I think it's gonna take a little more time than I wished it would.

**Comment by TheMisterFish** [comment]
@suerion could you try:
https://www.mediafire.com/file/qc4xo0sveu2ht84/offlineplayersreworked-1.0.9-bugfix.jar/file

I've tested that one on my own server first, should work fine.

**Comment by suerion** [comment]
[21:36:44] [main/INFO]: Hello from OfflinePlayersReworked!

will test it, server runs without issues yet :) will try on client

**Comment by suerion** [comment]
![2024-10-08_21 44 19](https://github.com/user-attachments/assets/2a2db3c4-6180-4409-8096-13cca728d915)

I think it work :)

**Comment by suerion** [comment]
Got only this issue on Server Console

```
[21:44:11] [Server thread/ERROR] (OfflinePlayersReworked) Cannot invoke "net.minecraft.class_2945.method_12778(net.minecraft.class_2940, Object)" because "this.field_6011" is null
 java.lang.NullPointerException: Cannot invoke "net.minecraft.entity.data.DataTracker.set(net.minecraft.entity.data.TrackedData, Object)" because "this.dataTracker" is null
        at net.rodofire.mushrooomsmod.entity.custom.PlotiEntity.initDataTracker(PlotiEntity.java:77) ~[mushrooomsmod-0.0.12-1.21.jar:?]
        at MC/net.minecraft.entity.Entity.<init>(MixinEntity.java:309) ~[server-intermediary.jar:?]
        at MC/net.minecraft.entity.LivingEntity.<init>(LivingEntity.java:270) ~[server-intermediary.jar:?]
        at MC/net.minecraft.entity.mob.MobEntity.<init>(MobEntity.java:148) ~[server-intermediary.jar:?]
        at MC/net.minecraft.entity.mob.PathAwareEntity.<init>(PathAwareEntity.java:17) ~[server-intermediary.jar:?]
        at MC/net.minecraft.entity.passive.PassiveEntity.<init>(PassiveEntity.java:29) ~[server-intermediary.jar:?]
        at MC/net.minecraft.entity.passive.AnimalEntity.<init>(AnimalEntity.java:42) ~[server-intermediary.jar:?]
        at net.rodofire.mushrooomsmod.entity.custom.PlotiEntity.<init>(PlotiEntity.java:34) ~[mushrooomsmod-0.0.12-1.21.jar:?]
        at MC/net.minecraft.entity.EntityType.create(EntityType.java:552) ~[server-intermediary.jar:?]
        at com.natamus.dailyquests_common_fabric.quests.types.TameAnimal.isAllowedIdentifier(TameAnimal.java:45) ~[dailyquests-1.21.1-1.8.jar:?]
        at com.natamus.dailyquests_common_fabric.quests.types.TameAnimal.getRandomQuestIdentifier(TameAnimal.java:53) ~[dailyquests-1.21.1-1.8.jar:?]
        at com.natamus.dailyquests_common_fabric.quests.functions.GenerateQuests.getNewQuests(GenerateQuests.java:122) ~[dailyquests-1.21.1-1.8.jar:?]
        at com.natamus.dailyquests_common_fabric.quests.functions.GenerateQuests.replaceAllPlayerQuests(GenerateQuests.java:80) ~[dailyquests-1.21.1-1.8.jar:?]
        at com.natamus.dailyquests_common_fabric.quests.functions.GenerateQuests.setInitialQuestScreen(GenerateQuests.java:23) ~[dailyquests-1.21.1-1.8.jar:?]
        at com.natamus.dailyquests_common_fabric.events.DailyQuestServerEvents.onEntityJoinLevel(DailyQuestServerEvents.java:46) ~[dailyquests-1.21.1-1.8.jar:?]
        at com.natamus.dailyquests.ModFabric.lambda$loadEvents$2(ModFabric.java:47) ~[dailyquests-1.21.1-1.8.jar:?]
        at net.fabricmc.fabric.api.event.lifecycle.v1.ServerEntityEvents.lambda$static$0(ServerEntityEvents.java:39) ~[fabric-lifecycle-events-v1-2.3.1+6c1df36019-a0320f6814921fbf.jar:?]
        at MC/net.minecraft.server.world.ServerWorld$ServerEntityHandler.handler$cpm000$fabric-lifecycle-events-v1$invokeEntityLoadEvent(ServerWorld.java:1730) ~[server-intermediary.jar:?]
        at MC/net.minecraft.server.world.ServerWorld$ServerEntityHandler.startTracking(ServerWorld.java:1611) ~[server-intermediary.jar:?]
        at MC/net.minecraft.server.world.ServerWorld$ServerEntityHandler.startTracking(ServerWorld.java:1570) ~[server-intermediary.jar:?]
        at MC/net.minecraft.server.world.ServerEntityManager.startTracking(ServerEntityManager.java:216) ~[server-intermediary.jar:?]
        at MC/net.minecraft.server.world.ServerEntityManager.addEntity(ServerEntityManager.java:186) ~[server-intermediary.jar:?]
        at MC/net.minecraft.server.world.ServerEntityManager.addEntity(ServerEntityManager.java:167) ~[server-intermediary.jar:?]
        at MC/net.minecraft.server.world.ServerWorld.addPlayer(MixinServerLevel.java:926) ~[server-intermediary.jar:?]
        at MC/net.minecraft.server.world.ServerWorld.onPlayerConnected(MixinServerLevel.java:912) ~[server-intermediary.jar:?]
        at MC/net.minecraft.server.PlayerManager.onPlayerConnect(MixinPlayerList.java:246) ~[server-intermediary.jar:?]
        at com.misterfish.patch.OfflinePlayer.createAndSpawnNewOfflinePlayer(OfflinePlayer.java:91) ~[offlineplayersreworked-1.0.9-bugfix-v2.jar:?]
        at com.misterfish.OfflinePlayersReworked.spawn(OfflinePlayersReworked.java:154) ~[offlineplayersreworked-1.0.9-bugfix-v2.jar:?]
        at com.misterfish.OfflinePlayersReworked.lambda$onInitializeServer$1(OfflinePlayersReworked.java:82) ~[offlineplayersreworked-1.0.9-bugfix-v2.jar:?]
        at com.mojang.brigadier.context.ContextChain.runExecutable(ContextChain.java:73) ~[brigadier-1.3.10.jar:?]
        at MC/net.minecraft.command.FixedCommandAction.execute(FixedCommandAction.java:29) ~[server-intermediary.jar:?]
        at MC/net.minecraft.command.FixedCommandAction.execute(FixedCommandAction.java:13) ~[server-intermediary.jar:?]
        at MC/net.minecraft.command.SourcedCommandAction.method_54405(SourcedCommandAction.java:8) ~[server-intermediary.jar:?]
        at MC/net.minecraft.command.CommandQueueEntry.execute(CommandQueueEntry.java:8) ~[server-intermediary.jar:?]
        at MC/net.minecraft.command.CommandExecutionContext.run(CommandExecutionContext.java:107) ~[server-intermediary.jar:?]
        at MC/net.minecraft.server.command.CommandManager.callWithContext(CommandManager.java:379) ~[server-intermediary.jar:?]
        at MC/net.minecraft.server.command.CommandManager.execute(CommandManager.java:309) ~[server-intermediary.jar:?]
        at MC/net.minecraft.server.network.ServerPlayNetworkHandler.executeCommand(ServerPlayNetworkHandler.java:1319) ~[server-intermediary.jar:?]
        at MC/net.minecraft.server.network.ServerPlayNetworkHandler.method_44356(ServerPlayNetworkHandler.java:1307) ~[server-intermediary.jar:?]
        at MC/net.minecraft.server.ServerTask.run(ServerTask.java:18) ~[server-intermediary.jar:?]
        at MC/net.minecraft.util.thread.ThreadExecutor.executeTask(ThreadExecutor.java:162) ~[server-intermediary.jar:?]
        at MC/net.minecraft.util.thread.ReentrantThreadExecutor.executeTask(ReentrantThreadExecutor.java:23) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.executeTask(MinecraftServer.java:864) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.executeTask(MinecraftServer.java:173) ~[server-intermediary.jar:?]
        at MC/net.minecraft.util.thread.ThreadExecutor.runTask(ThreadExecutor.java:136) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.runOneTask(MinecraftServer.java:846) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.runTask(MinecraftServer.java:840) ~[server-intermediary.jar:?]
        at MC/net.minecraft.util.thread.ThreadExecutor.runTasks(ThreadExecutor.java:145) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.runTasks(MinecraftServer.java:810) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.runTasksTillTickEnd(MinecraftServer.java:815) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:702) ~[server-intermediary.jar:?]
        at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:281) ~[server-intermediary.jar:?]
        at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
```

**Comment by suerion** [comment]
https://mclo.gs/KihM1gL

Here the log ;)

**Comment by TheMisterFish** [comment]
I'll have a look at it soon. Thanks for the log :)

**Comment by suerion** [comment]
> I'll have a look at it soon. Thanks for the log :)
> 
> 
> 
> By the way, could I ask you to test version 1.1.0 when I refactor some things? I don't like the fact I need this many imports.

Hey, yeah thats why i'm using my dev enviroment here in my pc. To test, fix and debug some issues.

You could send it, i will test, if it will work. You could also say, what i should look for. ;)

**Comment by suerion** [comment]
Hey, i think this is not really an issue, it is only an compatibilty problem with other mod, but it does nothing. It is only an logging error.

But if we could fix this, i think it could be fine. 

Will test after work again. Had an team meeting in 2 hours, hope it would be not that long

**Comment by TheMisterFish** [comment]
> Hey, i think this is not really an issue, it is only an compatibilty problem with other mod, but it does nothing. It is only an logging error.
> 
> But if we could fix this, i think it could be fine. 
> 
> Will test after work again. Had an team meeting in 2 hours, hope it would be not that long

The less error messages my mod causes, the happier I am 😇
Good luck with your meeting!

**Comment by suerion** [comment]
I think, it looks fine now :)

https://mclo.gs/DQHVQQ8


Only this comes from Automodpack ->
`[19:42:42] [Server thread/ERROR]: [OFF]Suerion isn't in the players map.`
https://github.com/Skidamek/AutoModpack/issues/292

Ignore Please the EOFException, this should be fixed in an update from an other mod :)

I think this are your Dev debug lines? https://mclo.gs/DQHVQQ8#L7140

**Comment by TheMisterFish** [comment]
I'll release 1.0.9 as soon as possible :)

Just to be sure, V3 gave no nullpointer exception, right?

**Comment by suerion** [comment]
Yeah, V3 give no errors

**Comment by TheMisterFish** [comment]
@suerion I tagged you in the other message as well but since you were not a part of that thread yet I'm not sure if the tagging worked correctly; if it did, sorry for the double-tagging.

1.0.9 has been released. Many thanks for your troubleshooting and log files, they were very handy.

I've created a 1.1.0-test.jar in https://github.com/TheMisterFish/Fabric_OfflinePlayersReworked/issues/34#issuecomment-2403144319
